.github/workflows/docker.yml

```
@@ -0,0 +1,84 @@
+name: Create and publish a Docker images
+
+on:
+  push:
+    tags:
+      - "v*.*.*"
+    branches:
+      - "master"
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-and-push-image:
+    name: PHP ${{ matrix.php-version }}, Fixer ${{ github.ref == 'refs/heads/master' && 'edge' || github.ref_name }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - php-version: '7.4'
+            php-suffix: '-php7.4'
+            alpine-version: '3.16'
+          - php-version: '8.0'
+            php-suffix: '-php8.0'
+            alpine-version: '3.16'
+          - php-version: '8.1'
+            php-suffix: '-php8.1'
+            alpine-version: '3.18'
+          - php-version: '8.2'
+            php-suffix: '-php8.2'
+            alpine-version: '3.18'
+          - php-version: '8.3'
+            php-suffix: '-php8.3'
+            alpine-version: '3.18'
+          # This is the latest supported PHP Version and will be provided without -phpX.X suffix
+          - php-version: '8.3'
+            php-suffix: ''
+            alpine-version: '3.18'
+
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@65b78e6e13532edd9afa3aa52ac7964289d1a9c1
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Make image lowercase
+        run: |
+          echo IMAGE_NAME_LOWER=$(echo ${{ env.IMAGE_NAME }} | tr '[:upper:]' '[:lower:]') >> ${GITHUB_ENV}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@9ec57ed1fcdbf14dcef7dfbe97b2010124a938b7
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME_LOWER }}
+          # "edge" will only build from master, "semver" only from tags
+          tags: |
+            type=edge,branch=master,suffix=${{ matrix.php-suffix }}
+            type=semver,pattern={{version}},suffix=${{ matrix.php-suffix }}
+            type=semver,pattern={{major}}.{{minor}},suffix=${{ matrix.php-suffix }}
+            type=semver,pattern={{major}},suffix=${{ matrix.php-suffix }}
+            type=semver,pattern=latest,suffix=${{ matrix.php-suffix }}
```

Following [this thread](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7555#discussion_r1430455616):

```suggestion
            type=edge,branch=master,suffix=php-${{ matrix.php-version }}
            type=semver,pattern={{version}},suffix=php-${{ matrix.php-version }}
            type=semver,pattern={{major}}.{{minor}},suffix=php-${{ matrix.php-version }}
            type=semver,pattern={{major}},suffix=php-${{ matrix.php-version }}
            type=semver,pattern=latest,suffix=${{ matrix.php-suffix }}
            type=raw,value=latest,enable=${{ '8.3' == matrix.php-version && 'tag' == github.ref_type }}
```

---

@hgiesenow I've now tested this on my fork and [it works better](https://github.com/Wirone/PHP-CS-Fixer/actions/runs/7277270285). Please apply changes done in [this commit](https://github.com/Wirone/PHP-CS-Fixer/commit/3debeb2b56bfc99b33be9464bb15c768749b2152) (I can't push changes into this PR because it's created from organisation).

---

FYI: I had to move `suffix` from `flavor` to `tags` again, because `latest` tag wasn't created (with or without `suffix=`). But if you can find a way to not repeat it everywhere while keeping `latest` tag, then great 🙂.

---

I've prepared suggestions to apply, feel free to do it via UI or on your side. I'm going to resolve this discussion as it's now redundant 🙂.